runtime/src/iree/hal/utils/caching_allocator.c

```diff
@@ -795,6 +795,10 @@ static iree_status_t iree_hal_caching_allocator_export_buffer(
                                           out_external_buffer);
 }
 
+bool iree_hal_caching_allocator_isa(iree_hal_allocator_t* base_value) {
```

Comment:
Yeah, this definitely won't work (if the wrapped buffer was an implementation detail maybe, but allocators are not). Design-wise, no one should have to know if a caching allocator is used.